openedx/core/lib/block_structure/block_structure.py

```diff
@@ -445,7 +445,7 @@ def __getitem__(self, usage_key):
         """
         Returns the BlockData associated with the given key.
         """
-        return self._block_data_map.get(usage_key)
+        return self._block_data_map[usage_key]
```

Comment:
**Note to reviewers:** I changed this code so it raises an error instead of returning `None` when a `usage_key` doesn't exist in the block structure.  This change surfaced 2 issues in tests that have now been fixed.